src/main/kotlin/com/vanniktech/maven/publish/BaseMavenPublishPlugin.kt

```diff
@@ -3,12 +3,18 @@ package com.vanniktech.maven.publish
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.api.tasks.Upload
+import org.gradle.util.VersionNumber
 
 internal abstract class BaseMavenPublishPlugin : Plugin<Project> {
 
   override fun apply(p: Project) {
     val extension = p.extensions.create("mavenPublish", MavenPublishPluginExtension::class.java, p)
 
+    val gradleVersion = VersionNumber.parse(p.gradle.gradleVersion)
+    if (gradleVersion < VersionNumber(MINIMUM_GRADLE_MAJOR, MINIMUM_GRADLE_MINOR, MINIMUM_GRADLE_MICRO, null)) {
+      throw IllegalArgumentException("You need gradle version 4.10.1 or higher")
+    }
+
     val pom = MavenPublishPom.fromProject(p)
     p.group = requireNotNull(pom.groupId) { "groupId is required to be set" }
     p.version = requireNotNull(pom.version) { "version is required to be set" }
```

Comment:
Can you remove these two lines? They shouldn't happen anymore.

Reply:
indeed, thanks